### ISSUE TITLE
Verifying existing JWT token

### ISSUE THREAD

**Issue by bluemoonljb** [lexik/LexikJWTAuthenticationBundle#990]
I am new to Lexik, we have an auth container that generate JWT token already, now in our new API project, we only need to verify the JWT but no need to generate it. 
Our JWT token is HS256, we don't set passphrase for the key, so passphrase is empty. 
In .env, our setting look like this: 

```
###> lexik/jwt-authentication-bundle ###
JWT_SECRET_KEY=%kernel.project_dir%/config/jwt/private.pem
JWT_PUBLIC_KEY=%kernel.project_dir%/config/jwt/public.pem
JWT_PASSPHRASE=
###< lexik/jwt-authentication-bundle ###
```

When we use the default jwt authenticator in security.yaml, it outputs ”invalid JWT Token", and I tracked the parse() method, and the exception is 
` exception: Unable to verify the given JWT through the given configuration. If the "lexik_jwt_authentication.encoder" encryption options have been changed since your last authentication, please renew the token. If the problem persists, verify that the configured keys/passphrase are valid.`

Anyone has idea what happened and how should we use JWT to authenticate existing JWT tokens? 
Thanks for any help in advance.

**Comment by fd6130** [comment]
~~You need to fill in the `JWT_PASSPHRASE` that you use to generate the JWT in `.env`. If both the passphrase are not the same, the JWT is not usable in your API.~~

**Comment by bluemoonljb** [comment]
@fd6130 Thanks for the response, but in our case, our key has no passphrase set when generating the key in the other auth container, so in this case, do I need to decode the JWT token by ourself? for example implement our own authenticate() method?

**Comment by fd6130** [comment]
> we have an auth container that generate JWT token already

Do you mean the JWT you have is not generate by this bundle?

> @fd6130 Thanks for the response, but in our case, our key has no passphrase set when generating the key in the other auth container, so in this case, do I need to decode the JWT token by ourself? for example implement our own authenticate() method?

My bad, now i think passphrase have nothing to do in your current situation. How about the secret_key and public_key? Are they the same as when you generate the JWT?

**Comment by bluemoonljb** [comment]
@fd6130 yes, the secret_key and public_key are copied from the other auth container, so they are the same.

**Comment by fd6130** [comment]
> @fd6130 yes, the secret_key and public_key are copied from the other auth container, so they are the same.

I guess it is signature algorithm issue. This bundle is using RS256 as default signature algorithm, is your other auth container using the same algorithm?

**Comment by bluemoonljb** [comment]
@fd6130  we use HS256 if I am correct.

**Comment by fd6130** [comment]
> @fd6130 we use HS256 if I am correct.

Ok, try add `encoder` config to `config/packages/lexik_jwt_authentication.yaml` :

```yaml
lexik_jwt_authentication:
  #...
  
  encoder:
      signature_algorithm: HS256
```

The encoder will now use HS256 algorithm to encode/decode the JWT.

**Comment by bluemoonljb** [comment]
@fd6130  I update as above, but since HS256 doesn't need private keys. I looked into the post https://github.com/lexik/LexikJWTAuthenticationBundle/issues/144

now my config in .env are these: 
```
JWT_SECRET_KEY=
JWT_PUBLIC_KEY=
JWT_PASSPHRASE=secretxxxxxx
```

```
lexik_jwt_authentication:
    secret_key: '%env(resolve:JWT_SECRET_KEY)%'
    public_key: '%env(resolve:JWT_PUBLIC_KEY)%'
    pass_phrase: '%env(JWT_PASSPHRASE)%'
    encoder:
        signature_algorithm: HS256
```


but I am still getting invalid token. any idea what could be the reason?

**Comment by fd6130** [comment]
I think the secret key is still needed for HS256. The passphrase and public key might not needed.

```
JWT_SECRET_KEY=secretxxxxxx
JWT_PUBLIC_KEY=
JWT_PASSPHRASE=
```

> HS256 (HMAC with SHA-256): A symmetric algorithm, which means that there is only one private key that must be kept secret, and it is shared between the two parties. Since the same key is used both to generate the signature and to validate it, care must be taken to ensure that the key is not compromised.

**Comment by bluemoonljb** [comment]
your suggestion works. now the custom authenticator is working fine. Thank you very much. @fd6130

**Comment by tomasz-kusy** [comment]
Sorry for digging up this issue, but I think it will be better than creating a new one.
I crashed on the same issue myself.
Before I found this explanation I didn't think it was necessary to set the encoder to a specific encoding type (HS256). In fact, that information is in the token. 
I have this situation: we want to use tokens with HS256/RS256 encoding both at the same time (different services).
Currently only the way the bundle is configured prevents us from doing this. As I understand there are no other objective reasons.
